beep/structure.py

```diff
@@ -152,21 +152,22 @@ def get_interpolated_steps(self, v_range, resolution, step_type='discharge', reg
             pandas.DataFrame: DataFrame corresponding to interpolated values.
         """
         if step_type is 'discharge':
-            group = self.data.groupby(["cycle_index", "step_index"]).filter(
-                determine_whether_step_is_discharging).groupby("cycle_index")
-            group.apply(lambda g: g[g['cycle_index'].isin(reg_cycles)])
+            step_filter = determine_whether_step_is_discharging
         elif step_type is 'charge':
-            group = self.data.groupby(["cycle_index", "step_index"]).filter(
-                determine_whether_step_is_charging).groupby("cycle_index")
-            group.apply(lambda g: g[g['cycle_index'].isin(reg_cycles)])
+            step_filter = determine_whether_step_is_charging
         else:
             raise ValueError("{} is not a recognized step type")
-
         incl_columns = ["current", "charge_capacity", "discharge_capacity",
                         "internal_resistance", "temperature", "cycle_index"]
         all_dfs = []
-        for cycle_index, df in tqdm(group):
-            new_df = get_interpolated_data(df, "voltage", field_range=v_range,
+        cycle_indices = self.data.cycle_index.unique()
```

Comment:
We might want to mention that we're purposefully not using the global groupby for the memory savings in a comment here, since we're doing something a bit non-idiomatic.  Something along the lines of "Use a cycle_index mask instead of a global groupby to save memory."